app/services/tsheets_service.rb

```diff
@@ -0,0 +1,29 @@
+class TsheetsService
+  def initialize
+    @adapter = TsheetsAdapter.new
+  end
+
+  def sync_users
+    unable_to_sync, synced = [], []
+    @tsheets_users = @adapter.find_users
+    @tsheets_users.each do |user|
+      result = sync_user(user)
+      unable_to_sync << user if result == :unable_to_sync
+      synced << user if result == :synced
+    end
+    {synced: synced, unable_to_sync: unable_to_sync}
+  end
+
+  private
+
+  def sync_user(tsheets_user)
+    return :nothing_done if User.find_by(tsheets_user_id: tsheets_user.id).present?
+    local_user = User.where('lower(first_name) = ? AND lower(last_name) = ?', tsheets_user.first_name.downcase, tsheets_user.last_name.downcase).first
```

Comment:
It's a good point. :-) I don't think we have any users with the same first and last name, though maybe it should just skip them if it finds multiple - this is not designed to catch everybody after all, just set up the users we can without too much trouble.